apps/python/web/routers/chat.py

```
@@ -81,14 +118,27 @@ def _sse_event(content: str, event: str | None = None) -> bytes:
+    session_file: Path,
+    env: dict[str, str],
+) -> None:
+    """Background task: drive the claude subprocess to completion and
+    append each stdout line into the job's replay buffer.
+
+    Always advances the job through running → done / failed regardless of
+    exception path. A grace-timer GC is scheduled at the end so memory
+    doesn't grow per-request.
+    """
+    chat_job_store.mark_running(job_id)
     proc = subprocess.Popen(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)